azurerm/internal/services/cosmos/resource_arm_cosmosdb_mongo_collection.go

```diff
@@ -103,12 +104,11 @@ func resourceArmCosmosDbMongoCollectionCreate(d *schema.ResourceData, meta inter
 				return fmt.Errorf("Error checking for presence of creating Cosmos Mongo Collection %s (Account %s, Database %s): %+v", name, account, database, err)
 			}
 		} else {
-			id, err := azure.CosmosGetIDFromResponse(existing.Response)
-			if err != nil {
+			if existing.ID != nil && *existing.ID != "" {
```

Comment:
we want to raise an error if there's already an existing resource (e.g. an ID is returned) so this wants to become:

```suggestion
			if existing.ID == nil || *existing.ID == "" {
```